src/osp/Active/SysNewton.cpp

```diff
@@ -9,33 +9,27 @@ using namespace osp::active;
 
 void cb_force_torque(const NewtonBody* body, dFloat timestep, int threadIndex)
 {
-    SysNewton* sysNewton = static_cast<SysNewton*>(
+    auto* sysNewton = static_cast<SysNewton*>(
```

Comment:
Static cast sounds risky.

Not dynamic cast?

Reply:
Newton objects allow a single custom 'userdata' void pointer to be stored in them. NewtonWorlds will always store a pointer to the SysNewton that initialized it, and all NewtonBody objects are created from an already-existing ACompNwtBody. There isn't any chance at this point that different types will be assigned as userdata. I've added a few comments in the latest commits to make this more clear. RTTI had also been avoided, and isn't used anywhere in the code.